bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -296,6 +297,11 @@ type User struct {
 	// SSHAuthorizedKeys specifies a list of ssh authorized keys for the user
 	// +optional
 	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+
+	// HashedPasswd specifies a hashed password for the user.
+	// NOTE: HashedPasswd can be used for existing and new users.
```

Comment:
If I got it right from the cloud-init codebase, HashedPasswd must be used only for existing users, or am I word

Also:
- could you kindly move this below Passwd, given that they are logically related
- should this be `changePasswd` or something that makes it more explicit its intended usage

@vincepri @sbueringer @enxebre @CecileRobertMichon opinions?

Reply:
I could only find the following doc:
```yaml
Most of these configuration options will not be honored if the user already exists. The following options are the exceptions; they are applied to already-existing users: plain_text_passwd, hashed_passwd, lock_passwd, sudo, ssh_authorized_keys, ssh_redirect_user.
```
> https://cloudinit.readthedocs.io/en/latest/topics/modules.html?highlight=hashed_passwd#users-and-groups

Based on that I would guess that they also work on new users, but it's not clear. 

Based on this comment https://github.com/kubernetes-sigs/cluster-api/pull/5885#discussion_r779262909
it works on new **and** existing comments (I assume this part was tested @ruifaling?)

Regarding naming, I was under the impression that we're trying to keep our fields roughly in sync with the field names of cloud-init, but I agree their name is not great. I assume the format of passwd and hashedpasswd in cloud-init is the same? If yes and if we don't necessarily want to mirror cloud-init names, +1 to `ChangePasswd`

Reply:
> HashedPasswd

The result of my test is that hashed_passwd could be used for new users and existing users

Reply:
> I could only find the following doc:
> 
> > Most of these configuration options will not be honored if the user already exists. The following options are the exceptions; they are applied to already-existing users: plain_text_passwd, hashed_passwd, lock_passwd, sudo, ssh_authorized_keys, ssh_redirect_user.
> > https://cloudinit.readthedocs.io/en/latest/topics/modules.html?highlight=hashed_passwd#users-and-groups
> 
> Based on that I would guess that they also work on new users, but it's not clear.
> 
> Based on this comment [#5885 (comment)](https://github.com/kubernetes-sigs/cluster-api/pull/5885#discussion_r779262909) it works on new **and** existing comments (I assume this part was tested @ruifaling?)
> 
> Regarding naming, I was under the impression that we're trying to keep our fields roughly in sync with the field names of cloud-init, but I agree their name is not great. I assume the format of passwd and hashedpasswd in cloud-init is the same? If yes and if we don't necessarily want to mirror cloud-init names, +1 to `ChangePasswd`

Yes ,I have tested it,hashedpasswd could be used for new users and existing users.
And yes, the format of passwd and hashedpasswd in cloud-init is the same, and the representative field of HashedPasswd in cloudinit is hashed_ passwd.
And HashedPasswd may be a little better than ChangePasswd based on the fact that it is suitable for both new users and existing users?

Reply:
@ruifaling given that hashed_passwd seems to work both for existing and new users, what if we keep only passwd and we output it as a hashed_passwd in the cloud init generated script (or we output it as both passwd and hased_passwd)? 
looping in @randomvariable and @yastij for their knowledge of cloud init

Reply:
One thing's for sure: hashed passwords won't be supportable on Windows since hashing is only done "server side", and looking at cloudbase-init, it therefore becomes (https://cloudbase-init.readthedocs.io/en/latest/userdata.html):

```
passwd - the user’s password. On Linux, the password is a hashed string, whereas on Windows the password is a plaintext string. If the password is not defined, a random password will be set.
```
with no support for `hashed_passwd`. Probably output as both for compatibility across both OS, with a comment that Windows won't accept the hashed version.

Ignition is the same as Cloud-Init and will handle all the same hashes, though there's no stated distinction between new and existing users: https://docs.fedoraproject.org/en-US/fedora-coreos/authentication/#_using_password_authentication

Reply:
To summarize. This PR only has to change `passwd` to `hashed_passwd` in the cloud init template?

Reply:
`hashed_passwd` will not work for Windows - so do that only if that's acceptable for that user group (it better be).

Reply:
> hashed_passwd will not work for Windows - so do that only if that's acceptable for that user group (it better be).

Okay, not sure I got that right :). We shouldn't break Windows user and I think that would be not acceptable for them. 

Just that I got it right:
* we need cloud-init passwd for Windows because hashed_passwd does not work
* we want cloud-init hashed_passwd for Linux users to be able to also set pw's of existing users.

Correct?

Reply:
It should probably be in a secret, especially if there is going to support a non hashed variant? That way it can be encrypted at rest.

Reply:
Ok, thanks all for the quick feedback.
I will try to reach out on people working on windows next week after MLK day, but 
WRT to "there is going to support a non hashed variant", this is not an option being considered in this PR

Reply:
@jayunit100 if you have some time could you give an opinion from windows land?